sycl/source/detail/memory_manager.cpp

```diff
@@ -176,27 +175,40 @@ void copyH2D(SYCLMemObjT *SYCLMemObj, char *SrcMem, QueueImplPtr SrcQueue,
   DstAccessRange[0] *= DstElemSize;
   SrcSize[0] *= SrcElemSize;
   DstSize[0] *= DstElemSize;
-
-  cl_command_queue CLQueue = UseExclusiveQueue
-                                 ? TgtQueue->getExclusiveQueueHandleRef()
-                                 : TgtQueue->getHandleRef();
-
+  
   if (1 == DimDst && 1 == DimSrc) {
     CHECK_OCL_CODE(clEnqueueWriteBuffer(
-        CLQueue, DstMem,
+        TgtQueue->getHandleRef(), DstMem,
         /*blocking_write=*/CL_FALSE, DstOffset[0], DstAccessRange[0],
         SrcMem + DstOffset[0], DepEvents.size(), &DepEvents[0], &OutEvent));
-  } else {
+    } else {
+
+    // test if the whole buffer is copied
+    if( SrcSize[0]   == SrcAccessRange[0] &&  SrcSize[1]   == SrcAccessRange[1]  &&  SrcSize[2]   == SrcAccessRange[2] &&
```

Comment:
if( SrcSize[0] -> if (SrcSize[0] .. please to keep it the same as the other coding style

Reply:
Why is it not just
```
if (SrcSize == SrcAccessRange && SrcSize == DstAccessRange
    && SrcOffset == sycl::id<3> { 0, 0, 0 } && SrcOffset == DstOffset) {
```
?
I put the comparison of the offset to 0 first to optimize the code.